autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -231,46 +237,54 @@ const Schedule: React.FC<RouteComponentProps> = () => {
     </div>
   ));
 
-  // build each day based on schedule
-  function getMeetingsForDay(day: number): Meeting[] {
-    // meetingDays = MTWRFSU
-    // day = MTWRF
-    return schedule.filter((meeting) => meeting.meetingDays[day]);
-  }
-  function getAvailabilityForDay(day: number): Availability[] {
-    return availabilityList.filter((avl) => avl.dayOfWeek === day);
-  }
-  function renderMeeting(meeting: Meeting): JSX.Element {
-    const colors = ['#500000', '#733333', '#966666', '#b99999', '#dccccc'];
-    return (
-      <MeetingCard
-        meeting={meeting}
-        bgColor={colors[uniqueSections.indexOf(meeting.section.id)]}
-        key={meeting.id}
-        firstHour={FIRST_HOUR}
-        lastHour={LAST_HOUR}
-      />
+  // let's see if useMemo reduces our render time
+  const meetingsForDays = React.useMemo(() => {
+    const uniqueSections = [...new Set(schedule.map((mtg: Meeting) => mtg.section.id))];
+    // build each day based on schedule
+    function getMeetingsForDay(day: number): Meeting[] {
+      // meetingDays = UMTWRFS
+      // day = MTWRF
+      return schedule.filter((meeting) => meeting.meetingDays[day]);
+    }
+    function renderMeeting(meeting: Meeting): JSX.Element {
+      const colors = [
+        '#500000', '#733333', '#966666', '#b99999',
+        '#871b1e', '#9a1d26', '#c2777d', '#9f494b', '#b76778',
+      ];
```

Comment:
Can you move this array to a separate file and export it? This would let us do a couple things:
- Prevent recreating the colors array every time meetings are rendered
- Use the array directly in the `uses unique colors for up to x different sections` test, which would let us assert that the are less unique sections than colors in the array instead of hardcoding the number
  - We could also check that all the background colors are in the array, but the only reason I would see this being helpful is if the the part in the test that calculates `meetingCards` somehow breaks at some point and selects incorrect elements so it's your call if you want to check this or not

Reply:
I elected not to implement the optional part of this change, as the computed style is an `rgb()` string and the provided styles are hex codes and it's too much of a hassle to conver between the two. I did, however, separate out the colors in their own file in the latest commit.